extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -0,0 +1,393 @@
+
+    this.userNotification_.registerUserNotification(this);
+
+    this.registerAction('dismiss', this.dismiss.bind(this));
+  }
+
+  /**
+   * @override
+   */
+  getId() {
+    return this.elementId_;
+  }
+
+  /**
+   * Executes a `POST` request to the url given on the `data-show-if-href`
+   * attribute.
+   * @param {string} ampUserId
+   * @return {!Promise<!PostResponseMetadata>}
+   * @private
+   */
+  postShowEndpoint_(ampUserId) {
+    // TODO: Determine wether this should be a POST with json body
+    // or a GET with data on the query params.
+    // see: https://github.com/ampproject/amphtml/issues/1228
+    this.ampUserId_ = ampUserId;
+    const getReq = {
+      method: 'POST',
+      credentials: 'include',
+      /** @type {PostRequestMetadata} */
+      body: {
+        'elementId': this.elementId_,
+        'ampUserId': ampUserId
+      }
+    };
+
+    return xhrFor(this.win_).fetchJson(this.showIfHref_, getReq);
+  }
+
+  /**
+   * Creates an POST to the specified `data-dismiss-href` url.
+   * @private
+   * @return {!Promise}
+   */
+  postDismissEnpoint_() {
+    return xhrFor(this.win_).fetchJson(this.dismissHref_, {
+      method: 'POST',
+      credentials: 'include',
+      body: {
+        'elementId': this.elementId_,
+        'ampUserId': this.ampUserId_
+      }
+    });
+  }
+
+  /**
+   * Success handler for `postShowEndpoint_`.
+   * @param {!PostResponseMetadata}
+   * @return {!Promise<boolean>}
+   * @private
+   */
+  onPostShowEndpointSuccess_(data) {
+    assert(typeof data['showNotification'] == 'boolean', '`showNotification` ' +
+        'should be a boolean. Got "%s" which is of type %s.',
+        data['showNotification'], typeof data['showNotification']);
+
+    if (data['showNotification']) {
+      return Promise.resolve(data['showNotification']);
+    }
+    // if no notification is shown, resolve right away with `false` since
+    // we dont need to show it.
+    return this.dialogResolve_(false);
+  }
+
+  /**
+   * Get async cid service.
+   * @return {!Promise}
+   * @private
+   */
+  getAsyncCid_() {
+    return cidFor(this.win_).then(cid => {
+      // `amp-user-notification` is our cid scope, while we give it a resolved
+      // promise for the 2nd argument so that the 3rd argument (the
+      // persistentConsent) is the one used to resolve getting
+      // the external CID.
+      return cid.get('amp-user-notification',
+          Promise.resolve(), this.dialogPromise_);
+    });
+  }
+
+  /** @override */
+  shouldShow() {
+    if (!this.getIfHrefPromise_) {
+      this.getIfHrefPromise_ = this.getAsyncCid_()
+          .then(this.postShowEndpoint_.bind(this))
+          .then(this.onPostShowEndpointSuccess_.bind(this))
+    }
+    return this.getIfHrefPromise_;
+  }
+
+  /**
+   * Shows the current user notification. Adds the `.amp-active` class to
+   * the element.
+   */
+  show() {
+    this.element.style.display = 'block';
+    this.element.classList.add('amp-active');
```

Comment:
Please don't switch to just `active` :) `amp-active` is totally fine. We don't want to collide with user-definable `active` classes that could mean absolutely anything. `amp-active` is already our canonical class for such cases elsewhere.